### PR TITLE
[Merged by Bors] - chore: make sure that some coercions have an attached definition

### PR DESCRIPTION
**Pull request by ADedecker**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 1]
```diff
@@ -72,13 +72,16 @@ instance (priority := 100) toLinearEquivClass (F R A B : Type*) [CommSemiring R]
...
+  { (f : A ≃ B), (f : A ≃+* B) with commutes' := commutes f }
+
 instance (F R A B : Type*) [CommSemiring R] [Semiring A] [Semiring B] [Algebra R A] [Algebra R B]
-    [_h : AlgEquivClass F R A B] : CoeTC F (A ≃ₐ[R] B) where
-  coe f :=
-    { (f : A ≃+* B) with
-      toFun := f
-      invFun := EquivLike.inv f
-      commutes' := AlgHomClass.commutes f }
+    [_h : AlgEquivClass F R A B] : CoeTC F (A ≃ₐ[R] B) :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Equiv.lean:83]:
```suggestion
    [AlgEquivClass F R A B] : CoeTC F (A ≃ₐ[R] B) :=
```

File: Mathlib/Algebra/Hom/NonUnitalAlg.lean [hunk 2]
```diff
@@ -93,13 +93,18 @@ variable [Semiring R] [NonUnitalNonAssocSemiring A] [Module R A]
 instance (priority := 100) {F : Type*} [NonUnitalAlgHomClass F R A B] : LinearMapClass F R A B :=
   { ‹NonUnitalAlgHomClass F R A B› with map_smulₛₗ := map_smul }
 
+/-- Turn an element of a type `F` satisfying `NonUnitalAlgHomClass F R A B` into an actual
+`NonUnitalAlgHom`. This is declared as the default coercion from `F` to `A →ₙₐ[R] B`. -/
+def toNonUnitalAlgHom {F R A B : Type*} [Monoid R] [NonUnitalNonAssocSemiring A]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Hom/NonUnitalAlg.lean:98]: `@[coe]`?

**Comment by eric-wieser** [comment]
bors d+

Thanks! It's perhaps worth checking the type arguments are all implicit to the new defs, if you haven't already.

File: Mathlib/Algebra/Hom/NonUnitalAlg.lean (hunk 2 above)
Comment (ADedecker) [reply Mathlib/Algebra/Hom/NonUnitalAlg.lean:98]: :facepalm:

**Comment by ADedecker** [comment]
bors merge

**Pull request closed without merging**